content/blog/graphql-as-a-gateway/index.md

```diff
@@ -0,0 +1,217 @@
+---
+title: GraphQL feat API Gateway
+date: '2019-07-01T08:00:00Z'
+cover: './cover.jpg'
+---
+
+![Photo by Christian Stahl on Unsplash](cover.jpg)
+
+GraphQL, REST, GRPC, Thrift... Have you ever imagined how to stick these together, in a micro-services architecture, and expose to the world? There are some common ways to do it, such as using Nginx or Kong, but an alternative way to do this is by using GraphQL in front of all services.
+
+## ⏩ API Gateway pattern -- a quick introduction
+
+Considering service A and B, how a client would be able to request it's data? The easiest and straight forward way would be to do a request to service A and another to B, each request pointing to different hosts (eg: `a.service/orders` and `b.service/users`).
+
+As the number of services grows, it is quite laborious to keep up with this strategy, as there could be too many services and requests to coordinate. To solve this, an API proxy can be used, where the client will request to only one service instead of multiple, and this proxy will orchestrate where this request should go, glueing all services in one place.
+
+While proxies just forward requests, API Gateways encapsulate more of the application internal architecture, working as a Facade, with some other responsibilities as well, such as:
+
+- Request/Response Transformation: requests made by a client can be reshaped before sent to internal services, with the same applying to responses
+- Request routing: as the proxy, route requests to specific services, translating to other protocols if required
+- Composition: one request to the gateway can actually be mapped to multiple internal API requests
+- Throttling: limit a user number of requests
```

Comment:
Limit the number of requests made by a user

Reply:
Updated with "Throttling: limit user requests up to a determined rate limit"